hummingbot/connector/exchange/huobi/huobi_exchange.py

```diff
@@ -0,0 +1,423 @@
+        {
+            "id": 59378,
+            "symbol": "ethusdt",
+            "account-id": 100009,
+            "amount": "10.1000000000",
+            "price": "100.1000000000",
+            "created-at": 1494901162595,
+            "type": "buy-limit",
+            "field-amount": "10.1000000000",
+            "field-cash-amount": "1011.0100000000",
+            "field-fees": "0.0202000000",
+            "finished-at": 1494901400468,
+            "user-id": 1000,
+            "source": "api",
+            "state": "filled",
+            "canceled-at": 0,
+            "exchange": "huobi",
+            "batch": ""
+        }
+        """
+        try:
+            exchange_order_id = await tracked_order.get_exchange_order_id()
+            if not exchange_order_id:
+                return None
+            path_url = CONSTANTS.ORDER_DETAIL_URL.format(exchange_order_id)
+            params = {
+                "order_id": exchange_order_id
+            }
+            update = await self._api_get(path_url=path_url, params=params, is_auth_required=True, limit_id=CONSTANTS.ORDER_DETAIL_LIMIT_ID)
+            return update
+        except Exception:
+            raise
+
+    async def _update_order_status(self):
+        tracked_orders = list(self.in_flight_orders.values())
+        for tracked_order in tracked_orders:
+            try:
+                order_update = await self.get_order_status(tracked_order)
+            except Exception:
+                self.logger().network(
+                    f"Error fetching status update for the order {tracked_order.client_order_id}: {order_update}.",
+                    app_warning_msg=f"Failed to fetch status update for the order {tracked_order.client_order_id}."
+                )
+                await self._order_tracker.process_order_not_found(tracked_order.client_order_id)
+                continue
+            if order_update is None:
+                continue
+            if order_update.get("status") != "ok":
+                self.logger().network(
+                    f"Error fetching status update for the order {tracked_order.client_order_id}: {order_update}.",
+                    app_warning_msg=f"Failed to fetch status update for the order {tracked_order.client_order_id}."
+                )
+                continue
+            order_state = order_update["data"]["state"]
+            # possible order states are "submitted", "partial-filled", "filled", "canceled"
+            if order_state not in ["submitted", "filled", "canceled", "partial-filled", "created", "partial-canceled", "canceling"]:
+                self.logger().debug(f"Unrecognized order update response - {order_update}")
+
+            # Calculate the newly executed amount for this update.
+            new_state = CONSTANTS.ORDER_STATE[order_state]
+            update = OrderUpdate(
+                client_order_id=tracked_order.client_order_id,
+                exchange_order_id=str(tracked_order.exchange_order_id),
```

Comment:
The exchange order id has to be taken from the status request result, not from the tracked order.